### ISSUE TITLE
Can't pass player to factory to be shared between nested directives

### ISSUE THREAD

**Issue by adamzerner** [brandly/angular-youtube-embed#45]
http://stackoverflow.com/questions/29472808/cant-pass-angular-youtube-embed-player-to-factory

**Comment by brandly** [comment]
Sorry, I'm on a phone right now, so I can't do a lot. It looks like you want a single player reference to be shared amongst various controllers or directives or what have you. 

If I understand correctly, you should try using a service instead of a factory, so that you'll have a singleton that can be passed about. 

Let me know if this helps. If not, could you supply a plunker or a jsfiddle? Thanks and best of luck

**Comment by adamzerner** [comment]
> It looks like you want a single player reference to be shared amongst various controllers or directives or what have you.

Correct.

> If I understand correctly, you should try using a service instead of a factory, so that you'll have a singleton that can be passed about.

I don't think the distinction between service and factory matters. From what I understand, both are singletons and the only difference is that services are instantiated using the `new` keyword. I found [this cheatsheet](https://dncmagazine.blob.core.windows.net/downloads/AngularCheatSheet-DNCMagazine.pdf) helpful.

I couldn't figure out how to include your library in the plunker, but my SO question should have enough information and I just edited it to include my code on GitHub.
